### ISSUE TITLE
Version equivalence

### ISSUE THREAD

**Issue by AniSkywalker** [keras-team/keras#9133]
**Version 1:**

`model = Sequential()

model.add(Embedding(vocab_size, emb_weights.shape[1], input_length=maxlen, weights=[emb_weights], trainable=trainable))

model.add(BatchNormalization(momentum=0.9))

model.add(Convolution1D(hidden_units, 2, kernel_initializer='he_normal', padding='valid',							activation='relu', input_shape=(1, maxlen), bias_initializer='he_normal'))
model.add(MaxPooling1D(pool_size=2))
model.add(Dropout(0.5))

model.add(Bidirectional(LSTM(hidden_units, kernel_initializer='he_normal', activation='sigmoid', dropout=0.5,bias_initializer='he_normal', recurrent_dropout=0.5, return_sequences=True,									unroll=True), merge_mode='sum'))

model.add(GlobalMaxPool1D())
model.add(Dropout(0.5))

model.add(Dense(2))
model.add(Activation('softmax'))

adam = Adam(lr=0.001)
model.compile(loss='categorical_crossentropy', optimizer=adam, metrics=['accuracy'])
print('No of parameter:', model.count_params())

print(model.summary())`
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
**Version 2:**

`input = Input(name='context', batch_shape=(None, maxlen))

emb = Embedding(vocab_size, emb_weights.shape[1], input_length=maxlen, weights=[emb_weights],					trainable=trainable)(input)

b_norm = BatchNormalization(momentum=0.9)(emb)

cnn1 = Convolution1D(hidden_units, 2, kernel_initializer='he_normal', padding='valid',						activation='relu', input_shape=(1, maxlen), bias_initializer='he_normal')(b_norm)

maxpool1 = MaxPooling1D(2)(cnn1)
dropout1 = Dropout(0.5)(maxpool1)

b_lstm1 = LSTM(hidden_units, kernel_initializer='he_normal', activation='sigmoid', return_sequences=True,	dropout=0.5, recurrent_dropout=0.5, bias_initializer='he_normal', unroll=True)(dropout1)

b_lstm2 = LSTM(hidden_units, kernel_initializer='he_normal', activation='sigmoid', return_sequences=True, dropout=0.5, recurrent_dropout=0.5, bias_initializer='he_normal', unroll=True,				go_backwards=True)(dropout1)

b_lstm = add([b_lstm1, b_lstm2])

gmax = GlobalMaxPool1D()(b_lstm)
gmax = Dropout(0.5)(gmax)

softmax = Dense(2, activation='softmax')(gmax)

model = Model(inputs=[input], outputs=softmax)

adam = Adam(lr=0.001)
model.compile(loss='categorical_crossentropy', optimizer=adam, metrics=['accuracy'])
print('No of parameter:', model.count_params())

print(model.summary())`

I have written two versions of the same code(I prefer the second style). I am getting different results. 2nd version is not converging. Am I doing it correctly? 

Input data size is: 
X.shape = (10000,50)
y.shape = (10000,2)

Thank you in advance.
